src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -113,12 +113,27 @@ class AuthScreens extends React.Component {
         // Load policies, maybe creating a new policy first.
         Linking.getInitialURL()
```

Comment:
Is `Linking.getInitialURL()` different than `ONYXKEYS.CURRENT_URL`?

Reply:
Good question. 'ONYXKEYS.CURRENT_URL` is set [here](https://github.com/Expensify/App/blob/cd397ae9aa6f7f2e2cb1910da3f1b28ce2147882/src/libs/Navigation/NavigationRoot.js#L54) in `NavigationRoot.parseAndStoreRoute` so I think I could use it instead. I'll try that out. Does anyone else know why we use `Linking.getInitialURL` in the first place?

Reply:
We are looking to see if we navigated here via an app link - which is what [`getInitialURL()`](https://reactnative.dev/docs/linking#getinitialurl) does.

That current url in Onyx is saved whenever there is a [navigation state change](https://reactnavigation.org/docs/navigation-container/#onstatechange) to track the focused screen

These are not really the same thing - but maybe the suggestion will work. I'd probably next look at the PR where we implemented this to see if there's any additional context.

Reply:
Here is [the commit where it was added](https://github.com/Expensify/App/pull/5706/commits/da3ea688a9936d918cd6b020f0ecbe952fd6d059) and [the PR](https://github.com/Expensify/App/pull/5706). I think `Linking.getInitialURL` is better than how it was done before this commit, but I think using ONYXKEYS.CURRENT_URL might also work. I will try it out.

Reply:
@roryabraham might remember if there was a reason

Reply:
I tried using `ONYXKEYS.CURRENT_URL` and it doesn't work very well because the navigation is ready before Onyx finishes setting the URL. I could wait until Onyx.set finished before setting the navigation to be ready, but that feels a bit hacky. Also, how would I construct the URL from the path? I had trouble accessing the base URL for the app through CONST or CONFIG.

I think I should stick with getInitialURL because we know it works.